app/policies/sign_comment_policy.rb

```diff
@@ -62,4 +62,15 @@ def sign_collaborator?
     record.sign.folders.left_outer_joins(:collaborations)
           .exists?(folders: { collaborations: { collaborator_id: user.id } })
   end
+
+  class Scope < Scope
+    def resolve
+      accessible_signs = SignPolicy::Scope.new(user, Sign).resolve
+      accessible_folders = FolderPolicy::Scope.new(user, Folder).resolve
+
+      scope
+        .where(sign_id: accessible_signs)
+        .where(folder_id: [nil, *accessible_folders])
```

Comment:
I imagine I'm probably just misunderstanding something here, but just double checking that there's not an instance where a sign is inside a folder, and the user has access to the specific sign, but not to the folder.

Reply:
I'll add a test for this

Reply:
OK, I've re-parsed this, and I don't think we need a test. This is a scope implementation, not a policy check. This means that it limits the records returned from the scope, rather than a binary result (authorized or not). 

In this case, this is expressing that the comments returned should be those on signs that the user has access to AND either comments without a folder (this means comments on the sign itself without a folder context), or comments in a folder that the user has access to. A user who doesn't have access to a folder shouldn't see those comments, even if they can access the sign.